src/tests/rollback-inconsistency.cc

```diff
@@ -0,0 +1,187 @@
+
+    You should have received a copy of the GNU Affero General Public License
+    along with PerconaFT.  If not, see <http://www.gnu.org/licenses/>.
+======= */
+
+#ident "Copyright (c) 2006, 2015, Percona and/or its affiliates. All rights reserved."
+
+#include "test.h"
+
+// insert enough rows with a child txn and then force an eviction to verify the rollback
+// log node is in valid state.
+// the test fails without the fix (and of course passes with it).
+// The test basically simulates the test script of George's. 
+
+static void
+populate(DB_ENV *env, DB *db) {
+    int r;
+    DB_TXN *parent = NULL;
+    r = env->txn_begin(env, NULL, &parent, 0); assert_zero(r);
+
+    DB_TXN *txn = NULL;
+    
+    r = env->txn_begin(env, parent, &txn, 0); assert_zero(r);
+
+    // populate 
+    for (int i = 0; i < 128; i++) {
+        int k = htonl(i);
+        char kk[4]; // 4 KB key
+        char str[220];
+        memset(kk, 0, sizeof kk);
+        memcpy(kk, &k, sizeof k);
+        memset(str,'a', sizeof str-1);
+        DBT key = { .data = kk, .size = sizeof kk };
+        DBT val = { .data = str, .size = 220 };
+        r = db->put(db, txn, &key, &val, 0);
+        assert_zero(r);
+    }
+
+    r = txn->commit(txn, 0); assert_zero(r);
+    
+    //we know the eviction is going to happen here and the log node of parent txn is going to be evicted
+    //due to the extremely low cachesize. 
+    
+    r = env->txn_begin(env, parent, &txn, 0); assert_zero(r);
+
+    // populate--- those rows will be lost in the rollback node of the parent txn
+    for (int i = 128; i < 256; i++) {
+        int k = htonl(i);
+        char kk[4]; // 4 KB key
+        char str[220];
+        memset(kk, 0, sizeof kk);
+        memcpy(kk, &k, sizeof k);
+        memset(str,'a', sizeof str-1);
+        DBT key = { .data = kk, .size = sizeof kk };
+        DBT val = { .data = str, .size = 220 };
+        r = db->put(db, txn, &key, &val, 0);
+        assert_zero(r);
+    }
+
+    r = txn->commit(txn, 0); assert_zero(r);
+    //again eviction due to the memory pressure. 256 rows is the point when that rollback log spills out. The spilled node
+    //will be written back but will not be dirtied by including rollback nodes from child txn(in which case the bug is bypassed).
+ 
+    r = env->txn_begin(env, parent, &txn, 0); assert_zero(r);
+
+    // populate
+    for (int i = 256; i < 512; i++) {
+        int k = htonl(i);
+        char kk[4]; // 4 KB key
+        char str[220];
+        memset(kk, 0, sizeof kk);
+        memcpy(kk, &k, sizeof k);
+        memset(str,'a', sizeof str-1);
+        DBT key = { .data = kk, .size = sizeof kk };
+        DBT val = { .data = str, .size = 220 };
+        r = db->put(db, txn, &key, &val, 0);
+        assert_zero(r);
+    }
+
+    r = txn->commit(txn, 0); assert_zero(r);
+    
+    r = parent->abort(parent); assert_zero(r);
+    //try to search anything in the lost range
+        int k = htonl(200);
```

Comment:
Why is this still indented?  Did you intend to perform more than one `db->get()` in a loop?